torch/_inductor/fx_passes/quantization.py

```diff
@@ -2007,10 +2109,49 @@ def _register_dequant_promotion():
                 with_reshape=input_dim_exceeds_two,
             ),
             pass_number=0,
+            extra_check_fn=_is_valid_general_dequant_promotion_pattern,
+            find_first_node_fn=_find_first_node_in_dequant_pattern,
+            end_node_target_candidates=[
+                aten.mul.Tensor,
+                prims.convert_element_type.default,
+                aten.reshape.default,
+            ],
             dtype=dtype,
         )  # pass_number=0 to run before weight prepack
 
 
+def _register_woq_mm_dequant_promotion():
+    # For a woq mm int8-mixed-bf16 dequant pattern, we should expect see the node list as:
+    # * aten.unsqueeze.default
+    # * prims.convert_element_type.default (to_fp32)
+    # * aten.reshape.default
+    def _find_first_node_in_dequant_pattern(_node):
+        if _node.target is aten.reshape.default:
+            return _node
+        else:
+            assert (
+                len(_node.args) >= 1
+            ), "In in dequant pattern, each node should have more than 1 arg."
+            return _find_first_node_in_dequant_pattern(_node.args[0])
+
+    _register_dequant_promotion_pass(
```

Comment:
Perhaps add comment to illustrate the pattern like what `_register_general_dequant_promotion` does?

Reply:
Thanks and added.